client/login/wp-login/index.jsx

```diff
@@ -118,12 +127,16 @@ class Login extends React.Component {
 				</a>;
 		}
 
-		const goBackLink = ! magicLoginView && <a
-			href="#"
-			key="back-link"
-			onClick={ this.goBack }>
-				<Gridicon icon="arrow-left" size={ 18 } /> { this.props.translate( 'Return' ) }
-			</a>;
+		let goBackLink;
+		if ( this.state.loaded && window.history.length > 1 ) {
+			goBackLink = ! magicLoginView && <a
```

Comment:
That's super minor and probably personal preference, but I feel like the test on `! magicLoginView`  should be part of the `if` condition above.